python/ray/serve/drivers.py

```diff
@@ -174,3 +180,79 @@ async def get_intermediate_object_refs(self) -> Dict[str, Any]:
     async def get_dag_node_json(self) -> str:
         """Returns the json serialized root dag node"""
         return self.dags[self.MATCH_ALL_ROUTE_PREFIX].dag_node_json
+
+
+@PublicAPI(stability="alpha")
+class ServegRPCIngress:
+    """
+    gRPC Ingress that starts gRPC server based on the port
+    """
+
+    def __init__(self, port):
+        self.server = grpc.aio.server()
+        self.port = port
+
+        def register_servicer():
+            """
+            protobuf Schema gRPC should generate bind function
+            (e.g. add_PredictAPIsServiceServicer_to_server) to bind gRPC server
+            and schema interface
+            """
+            bind_function_name = "add_{}_to_server"
+            module_name = self.__class__.__bases__[1].__module__
+            servicer_name = self.__class__.__bases__[1].__name__
+            getattr(sys.modules[module_name], bind_function_name.format(servicer_name))(
+                self, self.server
+            )
+
+        register_servicer()
+
+        self.setup_complete = asyncio.Event()
+        self.running_task = asyncio.get_event_loop().create_task(self.run())
+
+    async def run(self):
+        self.server.add_insecure_port("[::]:{}".format(self.port))
+        self.setup_complete.set()
+        await self.server.start()
+        await self.server.wait_for_termination()
+
+
+if not isinstance(serve_pb2_grpc.PredictAPIsServiceServicer, type):
+    # The reason we have this fake Driver class is that sphinx-build is to mock
```

Comment:
Is there a documentation link to this issue we can include here? I'm not familiar with the "metaclass conflict issue from multiple inheritance," and I don't think other developers will be either.

Reply:
It is a very normal issue when doing multiple inheritance after I did some research :)
http://www.phyast.pitt.edu/~micheles/python/metatype.html